File: api/historical-data.js
// Vercel Serverless Function - 历史数据API
export default async function handler(req, res) {
  // 设置CORS头
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    res.status(200).end();
    return;
  }

  try {
    const { symbol, timeframe = '1h', limit = 100 } = req.query;

    if (!symbol) {
      return res.status(400).json({
        success: false,
        message: '缺少symbol参数'
      });
    }

    // 生成模拟历史数据
    const generateOHLCVData = (count) => {
      const data = [];
      let basePrice = 45000;
      const now = Date.now();
      
      // 根据时间周期计算间隔
      const intervals = {
        '1m': 60 * 1000,
        '15m': 15 * 60 * 1000,
        '1h': 60 * 60 * 1000,
        '4h': 4 * 60 * 60 * 1000
      };
      
      const interval = intervals[timeframe] || intervals['1h'];

      for (let i = count - 1; i >= 0; i--) {
        const timestamp = now - (i * interval);
        const volatility = 0.02; // 2% 波动率
        
        const open = basePrice + (Math.random() - 0.5) * basePrice * volatility;
        const close = open + (Math.random() - 0.5) * open * volatility;
        const high = Math.max(open, close) + Math.random() * open * volatility * 0.5;
        const low = Math.min(open, close) - Math.random() * open * volatility * 0.5;
        const volume = Math.random() * 1000000;

        data.push({
          timestamp,
          open: parseFloat(open.toFixed(2)),
          high: parseFloat(high.toFixed(2)),
          low: parseFloat(low.toFixed(2)),
          close: parseFloat(close.toFixed(2)),
          volume: parseFloat(volume.toFixed(2))
        });

        basePrice = close; // 下一个K线的基准价格
      }

      return data;
    };

    const historicalData = generateOHLCVData(parseInt(limit));

    res.status(200).json({
      success: true,
      data: {
        symbol: symbol.toUpperCase(),
        timeframe,
        data: historicalData
      }
    });

  } catch (error) {
    console.error('Historical data API error:', error);
    res.status(500).json({
      success: false,
      message: '服务器内部错误'
    });
  }
}

File: api/historical-data.py
"""
Vercel Serverless Function - 历史数据API (Python版本)
"""

from http.server import BaseHTTPRequestHandler
import json
import urllib.parse
import random
import time
from datetime import datetime, timedelta

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        # 设置CORS头
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()

        try:
            # 解析URL参数
            parsed_url = urllib.parse.urlparse(self.path)
            query_params = urllib.parse.parse_qs(parsed_url.query)
            
            symbol = query_params.get('symbol', [None])[0]
            timeframe = query_params.get('timeframe', ['1h'])[0]
            limit = int(query_params.get('limit', [100])[0])
            
            if not symbol:
                response = {
                    'success': False,
                    'message': '缺少symbol参数'
                }
                self.wfile.write(json.dumps(response, ensure_ascii=False).encode('utf-8'))
                return

            # 生成模拟历史数据
            def generate_ohlcv_data(count):
                data = []
                base_price = 45000
                now = datetime.now()
                
                # 时间间隔映射
                intervals = {
                    '1m': timedelta(minutes=1),
                    '15m': timedelta(minutes=15),
                    '1h': timedelta(hours=1),
                    '4h': timedelta(hours=4)
                }
                
                interval = intervals.get(timeframe, intervals['1h'])
                
                for i in range(count - 1, -1, -1):
                    timestamp = now - (interval * i)
                    timestamp_ms = int(timestamp.timestamp() * 1000)
                    
                    volatility = 0.02  # 2% 波动率
                    
                    open_price = base_price + (random.random() - 0.5) * base_price * volatility
                    close_price = open_price + (random.random() - 0.5) * open_price * volatility
                    high_price = max(open_price, close_price) + random.random() * open_price * volatility * 0.5
                    low_price = min(open_price, close_price) - random.random() * open_price * volatility * 0.5
                    volume = random.uniform(100000, 1000000)

                    data.append({
                        'timestamp': timestamp_ms,
                        'open': round(open_price, 2),
                        'high': round(high_price, 2),
                        'low': round(low_price, 2),
                        'close': round(close_price, 2),
                        'volume': round(volume, 2)
                    })

                    base_price = close_price  # 下一个K线的基准价格
                
                return data

            historical_data = generate_ohlcv_data(limit)

            response = {
                'success': True,
                'data': {
                    'symbol': symbol.upper(),
                    'timeframe': timeframe,
                    'data': historical_data
                }
            }

            self.wfile.write(json.dumps(response, ensure_ascii=False).encode('utf-8'))

        except Exception as e:
            response = {
                'success': False,
                'message': f'服务器内部错误: {str(e)}'
            }
            self.wfile.write(json.dumps(response, ensure_ascii=False).encode('utf-8'))

    def do_OPTIONS(self):
        # 处理预检请求
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
